src/azul/terra.py

```diff
@@ -134,14 +134,19 @@ def parse(cls, source: str) -> 'TDRSourceName':
         else:
             assert False, type
         validate_uuid_prefix(prefix)
-        if partition_prefix_length == '':
+        try:
+            partition_prefix_length = int(partition_prefix_length)
+        except ValueError:
+            require(partition_prefix_length == '',
+                    'Must be number or empty',
+                    partition_prefix_length)
             partition_prefix_length = config.partition_prefix_length
-            source = f'{source}{partition_prefix_length}'
+            source += str(partition_prefix_length)
```

Comment:
I'm not convinced I made the right call here. Up to you whether to keep this version or revert to what you had originally.

Reply:
I preferred the old way personally. reverted.